From 02f6c8ce811eed267e0a04b895595b32d0e69b12 Mon Sep 17 00:00:00 2001
Message-Id: <02f6c8ce811eed267e0a04b895595b32d0e69b12.1417780878.git.chang-joon.lee@intel.com>
In-Reply-To: <cb098d33cdac3271103834fbce08218248d7bddb.1417780878.git.chang-joon.lee@intel.com>
References: <cb098d33cdac3271103834fbce08218248d7bddb.1417780878.git.chang-joon.lee@intel.com>
From: Durgadoss R <durgadoss.r@intel.com>
Date: Tue, 11 Nov 2014 17:55:10 +0530
Subject: [PATCH 18/34] FOR_UPSTREAM [VPG]: drm/i915: Disable DRRS when PSR is
 enabled

This patch prevents DRRS from initializing whenever PSR is
enabled on CHV platform.

Issue: GMINL-3876
Change-Id: Iad60f5cfab3fd40fb48f8a8e32610fec7794c149
Signed-off-by: Durgadoss R <durgadoss.r@intel.com>
---
 drivers/gpu/drm/i915/intel_dp.c |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index 60d91b3..d430e4b 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -5138,6 +5138,11 @@ void intel_dp_set_drrs_state(struct drm_device *dev, int refresh_rate)
 		return;
 	}
 
+	if (IS_CHERRYVIEW(dev) && i915.enable_psr) {
+		DRM_DEBUG_KMS("DRRS is disabled as PSR is enabled\n");
+		return;
+	}
+
 	if ((IS_CHERRYVIEW(dev) || INTEL_INFO(dev)->gen < 8)
 		&& intel_edp_is_psr_enabled(dev)) {
 		DRM_DEBUG_KMS("DRRS is disabled as PSR is enabled\n");
@@ -5230,6 +5235,11 @@ intel_dp_drrs_init(struct intel_digital_port *intel_dig_port,
 		return NULL;
 	}
 
+	if (IS_CHERRYVIEW(dev) && i915.enable_psr) {
+		DRM_DEBUG_KMS("DRRS disabled since PSR is enabled\n");
+		return NULL;
+	}
+
 	downclock_mode = intel_find_panel_downclock
 					(dev, fixed_mode, connector);
 
-- 
1.7.9.5

